File: frontend/src/types/document.ts
export type DocumentStatus = "ACTIVE" | "PENDING_DELETE" | "PENDING_REPLACE";

export type Document = {
  id: number;
  title: string;
  description: string;
  document_type: string;
  file_url: string;
  version: number;
  status: DocumentStatus;
  created_by: number;
  created_at: string;
};

File: backend copy/app/schemas/document.py
from datetime import datetime

from pydantic import BaseModel, Field

from app.core.enums import DocumentStatus
from app.schemas.common import PaginationMeta


class DocumentResponse(BaseModel):
    id: int
    title: str
    description: str
    document_type: str
    file_url: str
    version: int
    status: DocumentStatus
    created_by: int
    created_at: datetime

    class Config:
        from_attributes = True


class DocumentListResponse(BaseModel):
    items: list[DocumentResponse]
    meta: PaginationMeta


class DeleteRequestPayload(BaseModel):
    expected_version: int = Field(gt=0)
    note: str | None = Field(default=None, max_length=1000)
